test-data/unit/check-typeguard.test

```diff
@@ -370,3 +370,17 @@ if guard(a):
     reveal_type(a)  # N: Revealed type is "__main__.A"
 reveal_type(a)  # N: Revealed type is "__main__.A"
 [builtins fixtures/tuple.pyi]
+
+[case testTypeGuardNestedRestrictionAny]
+from typing_extensions import TypeGuard
+from typing import Any
+
+class A: ...
+def f(x: object) -> TypeGuard[A]: ...
+def g(x: object) -> None: ...
+
+def test(x: Any) -> None:
+    if not(f(x) or x):
+        return
+    g(x)
+[builtins fixtures/tuple.pyi]
```

Comment:
Can you also test `is_x(a) or is_y(a)` (two type guards) and type guard + isinstance (narrow down from e.g. `List[object]` to `List[str]` to test narrowing to a non-subtype with type guard in this context)?